docs-gb/models/securing-endpoints.md

```diff
@@ -0,0 +1,105 @@
+# Securing model endpoints
+
+In enterprise use cases, you may need to control who can access the endpoints for deployed models or pipelines. While Seldon Core 2 doesn’t natively provide this functionality, it integrates with various service meshes that support these requirements. Though Seldon Core 2 is service-mesh agnostic, the example on this page demonstrates how to set up authentication and authorization to secure a model endpoint using the Istio service mesh.
+
+## Securing Endpoints with Istio
+
+Service meshes offer a flexible way of defining authentication and authorization rules for your models. With Istio, for example, you can configure multiple layers of security within an Istio Gateway, such as a [TLS for HTTPS at the gateway](https://istio.io/latest/docs/tasks/traffic-management/ingress/secure-ingress/#configure-a-tls-ingress-gateway-for-a-single-host) level, [mutual TLS (mTLS) for secure internal communication](https://istio.io/latest/docs/tasks/traffic-management/ingress/secure-ingress/#configure-a-mutual-tls-ingress-gateway), as well as [AuthorizationPolicies](https://istio.io/latest/docs/reference/config/security/authorization-policy/) and [RequestAuthentication](https://istio.io/latest/docs/reference/config/security/request_authentication/) policies to enforce both authentication and authorization controls.
+
+## Prerequisites
+* [Deploy a model](/kubernetes/service-meshes/istio.md)
+* [Configure a gateway](/kubernetes/service-meshes/istio.md)
+* [Create a virtual service to expose the REST and gRPC endpoints](/kubernetes/service-meshes/istio.md)
+* Configure a OIDC provider to authenticate. Obtain the `issuer` url, `jwksUri`, and the `Access token` from the OIDC provider.
+{% hint style="info" %}
+**Note** There are many types of authorization policies that you can configure to enable access control on workloads in the mesh. 
+{% endhint %}
+
+In the following example, you can secure the endpoint such that any requests to the endpoint without the access token are denied.
+
+To secure the endpoints of a model, you need to:
```

Comment:
We should probably show how to do this in a general way, not for one model. In particular, I propose adding a VirtualService example that forwards the REST/gRPC endpoints for all models in a namespace (in `service-meshes/istio.md`)

Reply:
Hi @lc525 . Yes, this is added as a prerequisite ```* [Create a virtual service to expose the REST and gRPC endpoints](/kubernetes/service-meshes/istio.md)```

Reply:
I think that perhaps I was unclear in my comment. All the examples in `service-meshes/istio.md` target exposing REST/gRPC endpoints for **a particular model**. I was suggesting an example where we expose **all** the models loaded into a namespace via **one** virtual service.

This is so that for people having hundreds of models and want to add auth, we don't suggest "well, add hundreds of virtual services, and keep them updated as you add or remove models".

Reply:
Thank you @lc525 . I will work with Chandra to get an example and update it.

Reply:
@Rajakavitha1 I don't think that example can work, and even if it did, it doesn't solve  the initial point I've raised. It's possible I wasn't clear in what I was suggesting:

What I had in mind was to get towards something closer to this (untested!):
```
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: all-models-route
  namespace: seldon-mesh
spec:
  gateways:
  - istio-system/seldon-gateway
  hosts:
  - '*'
  http:
  - match:
    - uri:
        prefix: /v2
    name: models-http
    route:
    - destination:
        host: seldon-mesh.seldon-mesh.svc.cluster.local
  - match:
    - uri:
        prefix: /inference.GRPCInferenceService
    name: models-grpc
    route:
    - destination:
        host: seldon-mesh.seldon-mesh.svc.cluster.local
```

This is would be a catch-all VirtualService redirecting all REST requests coming with a /v2 path prefix (irrespective of the model) to our backend load balancer (named `seldon-mesh`), and similar for gRPC.

Could you, perhaps together with Chandra where help is needed, test that all the examples in this PR work as expected (or have they already been tested?)

Reply:
removing this section because @cherrymu  confirmed in the call that the customers have a system where a vs is created for their models.